File: data/code_pairs/fortran/703.f90
subroutine clusigt(n, x, incx, target, index, nclus)
    integer n, incx, target, nclus
    integer x(*), index(2,*)
    integer i, ib, ie, nb, ne
    if (n < 1) return
    if (incx >= 0) then
        ib = 1
        ie = 1
    else
        ib = 1 - incx * (n - 1)
        ie = 1 - incx * (n - 1)
    endif
    nb = 0
    do i = 1, n
        if (x(ib) > target) then
            if (i == 1 .or. x(ib-incx) <= target) then
                nb = nb + 1
                index(1,nb) = i
            endif
        endif
        ib = ib + incx
    end do
    ne = 0
    if (nb > 0) then
        do i = 1, n-1
            if (x(ie) > target .and. x(ie+incx) <= target) then
                ne = ne + 1
                index(2,ne) = i
            endif
            ie = ie + incx
        end do
        if (x(ie) > target) then
            ne = ne + 1
            index(2,ne) = n
        endif
    endif
    nclus = nb
end subroutine

! Test program for clusigt
program test_clusigt
    implicit none
    integer, allocatable :: x(:), index(2,:)
    integer :: nclus, i, n

    ! Example test case
    n = 7
    allocate(x(n))
    x = (/5, 6, 7, 8, 9, 10, 11/)
    allocate(index(2,n))
    call clusigt(n, x, 1, 9, index, nclus)
    
    print *, 'nclus =', nclus
    do i = 1, nclus
        print *, 'Cluster', i, ': Start =', index(1,i), ', End =', index(2,i)
    end do
end program

File: data/code_pairs/cpp/703.cpp
#include <vector>
#include <iostream>

void clusigt(int n, const std::vector<int>& x, int incx, int target, std::vector<std::vector<int>>& index, int& nclus) {
    if (n < 1) {
        nclus = 0;
        return;
    }

    int ib, ie, nb, ne;
    if (incx >= 0) {
        ib = 0;
        ie = 0;
    } else {
        ib = 1 - incx * (n - 1) - 1;
        ie = 1 - incx * (n - 1) - 1;
    }

    nb = 0;
    for (int i = 0; i < n; ++i) {
        if (x[ib] > target) {
            if (i == 0 || x[ib - incx] <= target) {
                index[0][nb] = i + 1; // Adjusting for one-based indexing
                nb++;
            }
        }
        ib += incx;
    }

    ne = 0;
    if (nb > 0) {
        for (int i = 0; i < n - 1; ++i) {
            if (x[ie] > target && x[ie + incx] <= target) {
                index[1][ne] = i + 1; // Adjusting for one-based indexing
                ne++;
            }
            ie += incx;
        }
        if (x[ie] > target) {
            index[1][ne] = n;
            ne++;
        }
    }

    nclus = nb;
}

int main() {
    std::vector<int> x = {5, 6, 7, 8, 9, 10, 11};
    int n = x.size();
    std::vector<std::vector<int>> index(2, std::vector<int>(n, 0));
    int nclus;

    clusigt(n, x, 1, 9, index, nclus);

    std::cout << "nclus = " << nclus << std::endl;
    for (int i = 0; i < nclus; ++i) {
        std::cout << "Cluster " << i+1 << ": Start = " << index[0][i] << ", End = " << index[1][i] << std::endl;
    }

    return 0;
}